File: binarySearch/binary_search.py
arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def binary_search(arr, target):
    high = len(arr)
    low = 0

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1


print(binary_search(arr, 1))

File: searchAlgos/binarySearch.js
const binarySearch = (arr, target) => {
  let high = arr.length - 1;
  let low = 0;

  while (low <= high) {
    const mid = Math.floor((low + high) / 2);
    if (target === arr[mid]) {
      return mid;
    }
    if (target < arr[mid]) {
      high = mid - 1;
    } else {
      low = mid + 1;
    }
  }
  return -1;
}

export default binarySearch;